Microsoft.Dynamics365.UIAutomation.Browser/Extensions/SeleniumExtensions.cs

```diff
@@ -588,8 +588,15 @@ public static IWebElement WaitUntilAvailable(this IWebDriver driver, By by, Time
 
             try
             {
-                returnElement = wait.Until(d => d.FindElement(by));
-
+                var foundElements = wait.Until(d => d.FindElements(by));
```

Comment:
I suggest not to return success = true; if wait.Until return null or an empty collection.
In such case failiureCallback is also not called, eben if the element was not found.
Hier my code:
```csharp
                int? elementsCount = foundElements?.Count;
                if (elementsCount > 0)
                {
                    returnElement = elementsCount > 1 
                        ? foundElements.FirstOrDefault(x => x.Displayed) 
                        : foundElements[0];
                    success = true;
                }```
There is also other optimizations posible ex.: 
1. success can be bool instead bool?.
2. catch (NoSuchElementException) is not required, it is an ignoredException

Reply:
Hi AngelRodriguez8008,
Sorry I haven't been checking github and i've moved on from doing automated testing for D365. 
Your previous post showed great explanation to the root of the issue. Can you please refactor the codes and apply the fix for it because you can test the login dialog issue. I did have some random UCI login issue few weeks ago but it was inconsistent (1 issue out of 5 runs sort of thing) so I didn't look into it. 
If you want me to do it, i'll need to spend sometime setting up D365 classic web app and test the codes. Let me know what you wanna do. Cheers

Reply:
Hi @KhoaAN, I already fixed it in two PRs (#735, #747)